backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/impl/ElasticsearchClientFactoryImpl.java

```diff
@@ -246,6 +254,15 @@ private HttpAsyncClientBuilder customizeHttpClientConfig(HttpAsyncClientBuilder
 		for ( ElasticsearchHttpClientConfigurer configurer : configurers ) {
 			configurer.configure( clientConfigurationContext );
 		}
+		if ( customConfig.isPresent() ) {
+			BeanHolder<? extends ElasticsearchHttpClientConfigurer> customConfigBeanHolder = customConfig.get();
+			try {
+				customConfigBeanHolder.get().configure( clientConfigurationContext );
+			}
+			finally {
+				customConfigBeanHolder.close();
```

Comment:
You have no guarantee that this code will actually be called. If something fails before the callback is called, this finally block won't be called.

The `try`/`finally` really should be in `createClient()`.